### PR TITLE
Bluetooth: audio: has: Fix building with preset support disabled

### PR DESCRIPTION
**Pull request by MariuszSkamra**
This fixes regression causing compilation errors seen when the
code is built without preset support (BT_HAS_PRESET_COUNT = 0).

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

### CODE REVIEW

**Review by Thalley — CHANGES_REQUESTED** [review]
Minor request: The commit message should tell why the Kconfig dependency was removed

File: subsys/bluetooth/audio/Kconfig.has [hunk 1]
```diff
@@ -41,7 +41,7 @@ config BT_HAS_PRESET_NAME_DYNAMIC
 
 config BT_HAS_PRESET_CONTROL_POINT_NOTIFIABLE
 	bool "Preset Control Point Notifiable support"
-	depends on BT_HAS_PRESET_SUPPORT && BT_EATT
+	depends on BT_EATT
```
[the remark this answers is not in the document]
Comment (Thalley) [reply subsys/bluetooth/audio/Kconfig.has:44]: Ah - Completely missed that - Thanks

**Review by Thalley — DISMISSED** [review]

File: samples/bluetooth/hap_ha/sample.yaml [hunk 2]
```diff
@@ -7,6 +7,14 @@ tests:
     platform_allow: native_posix
     tags: bluetooth
     build_only: true
+  sample.bluetooth.hap_ha.monaural_no_presets:
+    harness: bluetooth
+    extra_args: OVERLAY_CONFIG="binaural.conf"
```
Comment (MariuszSkamra) [inline samples/bluetooth/hap_ha/sample.yaml:12]: This one needs to be removed

**Review by Thalley — APPROVED** [review]

**Review by szymon-czapracki — APPROVED** [review]
LGTM

**Pull request merged**